### PR TITLE
FRO-40: spell simulator

### PR DESCRIPTION
**Pull request by tomeq93**

### CODE REVIEW

**Review by hexonaut — CHANGES_REQUESTED** [review]
This looks good. Some small fixes:

 * Should explicitly set the rpc url on all foundry commands. I have `ETH_RPC_URL` set in my own environment which will override the assumption of localhost you're making. I would either set this `ETH_RPC_URL` env var in the script or provide it as an argument to all commands.
 * Newlines at the end of each file.

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,5 @@ reports/*
 !reports/.empty
 
 node_modules/
+
+.DS_Store
```
Comment (hexonaut) [inline .gitignore:22]: Be careful not to commit OS files like this. I recommend having a `.gitignore` in your home directory for stuff like this:

```
.DS_Store
*.sublime-project
*.sublime-workspace
*.code-workspace
.vscode/
lcov.info
```

File: scripts/fork-and-cast-spell.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -e
+
+trap "kill 0" EXIT
```
Comment (hexonaut) [inline scripts/fork-and-cast-spell.sh:4]: You should add `unset ETH_FROM` as well here. I have this environment variable set and it is messing with the submission.

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: scripts/fork-and-cast-spell.sh (hunk 2 above)
Comment (tomeq93) [reply scripts/fork-and-cast-spell.sh:4]: 👍 --from passed to avoid issues with different ETH_FROM & private key

File: .gitignore [hunk 3]
```diff
@@ -17,4 +17,4 @@ docs/
 reports/*
 !reports/.empty
 
-node_modules/
+node_modules/
```
Comment (krzkaczor) [inline .gitignore:20]: can we get rid of any changes to this file?

**Review by krzkaczor — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (tomeq93) [reply .gitignore]: 👍 done

**Review by hexonaut — APPROVED** [review]
ok this is good going to merge into parent PR

**Pull request merged**